config/v1/types_project.go

```diff
@@ -20,8 +19,13 @@ type Project struct {
 }
 
 type ProjectSpec struct {
-	// project request message
-	// project request template
+	// ProjectRequestMessage is the string presented to a user if they are unable to request a project via the projectrequest api endpoint
+	ProjectRequestMessage string `json:"projectRequestMessage"`
+
+	// ProjectRequestTemplate is the template to use for creating projects in response to projectrequest.
+	// This must point to a template in 'openshift-config' namespace. It is optional.
+	// If it is not specified, a default template is used.
+	ProjectRequestTemplateName string `json:"projectRequestTemplateName"`
```

Comment:
Needs to be a struct like `ConfigMapNameReference` so we have quality openapi.

Reply:
`projectRequestTemplate`

Reply:
Long term I am curious how this relates to https://github.com/kubernetes/enhancements/pull/645

Reply:
Please try to mimic the wording I have with `ConfigMapNameReference` (and its usages).  Consistency is helpful.